disk collection browser updates

dcaputi1 opened the pull request:
The initial release of the Disk Collection Browser was not completely tested and lacked the ability to convert between artwork data sources. There's 3 schemes that involve persisting artwork data and one that uses file naming. The file naming scheme stands alone and can not be converted to/from the other 3 schemes. 

The 3 storage schemes are selected in the RespeQt options dialog under the Disk Collection Browser tab/page. These are the 3 radio buttons on the top of the page. The details on these options are described in the .rtf document found in the deployed app folder.

This commit allows the user to create a working definition of titles/labels and cover art then later select a different storage scheme in the options dialog.  When the new selection is saved, the sytem will automatically convert the persisted data from one to the other.

dcaputi1 committed:
index column added to diskbrowser list

- added cclumn 0 for optional index
- sort disk list by index

dcaputi1 committed:
moved dbsettings and fixed crash

- dbsettings moved from dialog to respeqtsettings
- fixed crash when db source type changed then used without reset

dcaputi1 committed:
WIP: dcb - convert between the 3 setting sources

Disk Collection Browser: convert to/from all 3 DB source types if changed in the settings dialog.
- commit implements to/from default app settings (reg on win) and app data json file

dcaputi1 committed:
WIP: dcb - from app settings to sub-dir

- convert from app settings (ini) to sub-dirs with seperate json

dcaputi1 committed:
WIP: dcb - single to multi json

- convert from db source 3 to 2
- don't convert with dcb open, popup warn

dcaputi1 committed:
persist options dialog geometry

josch1710 commented on sources.pri:
```diff
@@ -79,6 +79,7 @@ SOURCES += \
     src/diskbrowser/diskbrowserdlg.cpp
 
 HEADERS += \
+    $$PWD/include/diskbrowser/diskbrowser.h \
```
Is the $$PWD needed?

dcaputi1 replied:
I may have used a QtCreator wizard to generate the new header file which is why it was only added to the .pri file at the top with the unneeded $$PWF.
Good catch! ... fixed.

josch1710 reviewed (commented):
You added a file to sources.pri, but not to CMakeFiles.txt

dcaputi1 committed:
added diskbrowser.h to CMakeLists.txt

This header was added as a convenience file so that consumers of diskbrowser functions can simply include the one header. I may have used a QtCreator wizard to generate the file which is why it was added to the .pri file at the top with the unneeded $$PWD.

dcaputi1 commented:
For some reason, I am no longer able to build using CMake on my Windows PC.
Let me know if it's not working and I'll see if I can figure out what changed.  I'm pretty sure I was able to successfully build and test my previous PR with CMake.

josch1710 reviewed (approved):
Looks good.
